src/main/java/TTT/databaseUtils/CustomUserDAO.java

```diff
@@ -0,0 +1,70 @@
+package TTT.databaseUtils;
+
+import TTT.users.CustomUser;
+import jakarta.persistence.NoResultException;
+import jakarta.persistence.criteria.CriteriaBuilder;
+import jakarta.persistence.criteria.CriteriaDelete;
+import jakarta.persistence.criteria.CriteriaQuery;
+import jakarta.persistence.criteria.Root;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+
+public class CustomUserDAO {
+
+    private final BCryptPasswordEncoder passwordEncoder = new BCryptPasswordEncoder();
+    private final SessionFactory sessionFactory = UserSessionFactory.getUserSessionFactory();
+
+    public boolean saveUser(CustomUser customUser) {
+        if (findCustomUser(customUser.getEmail()) == null) {
+            Session session = sessionFactory.openSession();
+            Transaction transaction = session.beginTransaction();
+            customUser.setPassword(passwordEncoder.encode(customUser.getPassword()));
+            session.merge(customUser);
+            transaction.commit();
+            session.close();
+            return true;
+        } else {
+            System.out.println("EntityExistsException: email alerady exist.");
```

Comment:
Better to throw an exception, you can't do anything with console print

Reply:
even if I want to add a user by manually setting his ID, hibernate will change it, not allowing duplication. Therefore, No entityException occurs. Print, however, I simply copied it to know exactly what I mean.